minidump-processor/src/op_analysis.rs

```diff
@@ -153,62 +154,174 @@ mod amd64 {
+            self.indirect(&mut memory_accesses, &decoded_instruction)?;
+            Ok(memory_accesses)
+        }
+
+        fn direct(
+            &self,
+            accesses: &mut Vec<MemoryAccess>,
+            decoded_instruction: &Instruction,
+        ) -> Result<(), OpAnalysisError> {
+            // Shortcut -- If the instruction doesn't access memory, just return an empty list
+            let mem_size = match decoded_instruction.mem_size() {
+                Some(access) => access.bytes_size(),
+                None => return Ok(()),
             };
 
+            for idx in 0..decoded_instruction.operand_count() {
+                let operand = decoded_instruction.operand(idx);
+
+                let maybe_access = match operand {
+                    Operand::DisplacementU32(disp) => Some(MemoryAccess {
+                        address: disp.into(),
+                        is_likely_null_pointer_dereference: false,
+                        size: mem_size,
+                    }),
+                    Operand::DisplacementU64(disp) => Some(MemoryAccess {
+                        address: disp,
+                        is_likely_null_pointer_dereference: false,
+                        size: mem_size,
+                    }),
+                    other_operand => {
+                        if let Some(op_info) = RegOperandInfo::try_from_operand(other_operand) {
+                            Some(self.calculate_address(mem_size, op_info)?)
+                        } else {
+                            None
+                        }
+                    }
+                };
+
+                if let Some(access) = maybe_access {
+                    accesses.push(access);
+                }
+            }
+
+            Ok(())
+        }
+
+        /// Indirect memory accesses include instructions which change the instruction pointer.
+        fn indirect(
+            &self,
+            accesses: &mut Vec<MemoryAccess>,
+            decoded_instruction: &Instruction,
+        ) -> Result<(), OpAnalysisError> {
+            match decoded_instruction.opcode() {
+                Opcode::CALL | Opcode::CALLF | Opcode::JMP | Opcode::JMPF | Opcode::JMPE => {
+                    if decoded_instruction.operand_count() != 1 {
+                        tracing::warn!("call/jmp instruction had incorrect operand count");
+                        return Ok(());
+                    }
+                    // We assume that relative offsets (for CALL, JMP) and absolute values (CALLF,
+                    // JMPF) will be valid, so we don't check immediate operands, only registers.
+                    if let Operand::Register(reg) = decoded_instruction.operand(0) {
+                        let address = self.get_reg(reg)?;
+                        accesses.push(MemoryAccess {
+                            address,
+                            is_likely_null_pointer_dereference: address == 0,
+                            size: Some(1),
```

Comment:
I wasn't sure whether this was appropriate or whether `None` would be better, since we don't know the actual read size.

Reply:
It's fine! Since x86 ISA variable-sized the first access by a jump instruction can be as small as one byte.